src/spotify_to_tidal/__main__.py

```diff
@@ -24,13 +25,16 @@ def main():
         spotify_playlist = spotify_session.playlist(args.uri)
         tidal_playlists = _sync.get_tidal_playlists_dict(tidal_session)
         tidal_playlist = _sync.pick_tidal_playlist_for_spotify_playlist(spotify_playlist, tidal_playlists)
-        _sync.sync_list(spotify_session, tidal_session, [tidal_playlist], config)
+        _sync.sync_playlists_wrapper(spotify_session, tidal_session, [tidal_playlist], config)
     elif config.get('sync_playlists', None):
         # if the config contains a sync_playlists list of mappings then use that
-        _sync.sync_list(spotify_session, tidal_session, _sync.get_playlists_from_config(spotify_session, tidal_session, config), config)
+        _sync.sync_playlists_wrapper(spotify_session, tidal_session, _sync.get_playlists_from_config(spotify_session, tidal_session, config), config)
     else:
         # otherwise just use the user playlists in the Spotify account
-        _sync.sync_list(spotify_session, tidal_session, _sync.get_user_playlist_mappings(spotify_session, tidal_session, config), config)
+        _sync.sync_playlists_wrapper(spotify_session, tidal_session, _sync.get_user_playlist_mappings(spotify_session, tidal_session, config), config)
+
+    if not args.disable_favorites_sync:
+        _sync.sync_favorites_wrapper(spotify_session, tidal_session, config)
```

Comment:
We should probably only sync favorites when syncing the full collection (i.e. move this up slightly to the `else` block)

Reply:
But then a user would only be able to sync the favorites when running a full library sync. I think changing the flag to `enable-favorites-sync` would be the better compromise.

Reply:
I'd like to have the following behavior:
* Calling script with no arguments: Sync favorites by default after completing other playlists, unless `--disable-sync-favorites` provided
* Calling script with explicit uri or config mapping: Never syncs favorites
* Calling script with `--sync_favorites`: Syncs favorites only

Reply:
See the patch file I included [above](https://github.com/spotify2tidal/spotify_to_tidal/pull/55#discussion_r1631793546)

Reply:
I left some TODO's there to use a subparser, so if you feel like addressing the todo's it would be appreciated